### ISSUE TITLE
reorganize status page to clearly show SLAs for delivery and publishing

### ISSUE THREAD

**Issue by rofe** [adobe/helix-home#23]
We decided during the [Basel hackathon](https://github.com/adobe/helix-home/blob/master/hackathons/5-bsl.md) that the Helix SLAs are defined for delivery (goal: 99.99%) and  publishing (goal: 99.9%). In oder to be able to see how we are tracking against these goals, we need to rearrange the components on https://status.project-helix.io into new component groups for delivery and publishing.

**Comment by rofe** [comment]
Proposal:
- Delivery
  - Helix Homepage
  - Helix Embed
  - Helix Static
- Publishing
  - Helix Publish
- Other Helix Services
  - Helix Logging
  - Helix Performance
  - Helix Bot

The question is how we should categorize and measure the 3rd party services. Apart from CircleCI, they all affect both Delivery and Publishing. Should we duplicate the components, or continue to list them in separate component groups?

**Comment by kptdobe** [comment]
While working on https://github.com/adobe/project-helix/pull/403, I realised 2 things:

1. we should agree on the terminology (delivery vs production, publishing  vs deployment, ...)
2. some services play a different role and have a different importance depending on which side of the architecture they are considered (same as your `they all affect both Delivery and Publishing`). This leads to a multi-dimensional matrix which is hard to represent.

In general, I think the status should reflect both:

1. what is "business" critical
2. what we can do something about

Piling a long list of statuses will be useless on the long run, we must categorise them by severity / criticality and also have the on-call / escalation process setup only for the "business" critical services.

Example: the Helix Bot. The runbook says `Don't do anything yet, the component is still under development and fails frequently.` then it has nothing to do in the "business" critical bucket and should not wake us up if it fails.

**Comment by rofe** [comment]
> 1. we should agree on the terminology (delivery vs production, publishing  vs deployment, ...)

Agreed. Let's use _Delivery_ for anything that affects the end-users of our customers, and _Publishing_ for anything that affects the process for our customers to put content live.

> 2. some services play a different role and have a different importance depending on which side of the architecture they are considered (same as your `they all affect both Delivery and Publishing`). This leads to a multi-dimensional matrix which is hard to represent.

Yep, I'm still unsure about how to reflect that in the status page. I wonder if you can have multiple 3rd party components like Fastly, GitHub on the same status page.

> In general, I think the status should reflect both:
> 
> 1. what is "business" critical
> 2. what we can do something about
> (...)

Fully agreed. I tried to do that by moving critical components into the _Delivery_ and _Publishing_ buckets, and putting anything else into the _Other Helix Services_ bucket (we could rename the latter to _Non-critical ..._ to be more specific). We should still keep two distinct buckets for critical components because we agreed on different SLAs for _Delivery_ and _Publishing_, and I still think that makes sense. The bucket will aggregate the overall uptime for easy tracking.

**Comment by trieloff** [comment]
In Pingdom, I have tagged the services with `delivery`, `publishing`, and `development`, which I believe correctly represents the severities. When `delivery` is down, your website can't be visited. When `publishing` is down, content updates won't go through and when development` is down, you can't roll out new code.

We can use this in PagerDuty to have different (slower) escalation policies for less critical services and we can also use this to have better automated status messages on StatusPage.

When it comes to calculating the SLA (and displaying it on Statuspage), I'd like to check if we can achieve this without having to regroup the services and instead use a custom metric like this: https://help.statuspage.io/help/introduction-1 That way, we could have three metrics and use Pingdom's aggregation instead of Statuspage's.

**Comment by rofe** [comment]
Thanks for the additional input and ideas.

> In Pingdom, I have tagged the services
Where can I see those tags, or search by them?

> we could have three metrics and use Pingdom's aggregation
Looks to me like Statuspage currently only supports two metrics from Pingdom: `uptime` and `responsetime`
